requirements-docs.txt

```diff
@@ -1,3 +1,7 @@
 mkdocs-material==9.4.6
 mkdocs-awesome-pages-plugin==2.9.2
 mkdocs-git-revision-date-localized-plugin==1.2.1
+mkdocs-macros-plugin
+mkdocstrings[python]
+mkdocs-gen-files
+mkdocs-section-index
```

Comment:
I would suggest freezing the dependencies so we don't get accidentally broken docs.